.kokoro/build.sh

```diff
@@ -23,24 +27,44 @@ bundle update
 
 # CHANGED_DIRS is the list of top-level directories that changed. CHANGED_DIRS will be empty when run on master.
 # See https://github.com/GoogleCloudPlatform/google-cloud-python/blob/master/.kokoro/build.sh for alt implementation
-CHANGED_DIRS=$(git --no-pager diff --name-only HEAD $(git merge-base HEAD master) | grep "/" | cut -d/ -f1 | sort | uniq || true)
+CHANGED_DIRS="$(git --no-pager diff --name-only HEAD $(git merge-base HEAD master) | grep "/" | cut -d/ -f1 | sort | uniq || true)"
+
+GEMSPECS=($(git ls-files -- */*.gemspec | cut -d/ -f1))
+UPDATED_GEMS=()
+RUBY_VERSIONS=("2.3.7" "2.4.4" "2.5.1")
+
+for i in "${GEMSPECS[@]}"; do
+  for j in "${CHANGED_DIRS[@]}"; do
+    if [ "$i" = "$j" ]; then
+      UPDATED_GEMS+=($i)
+    fi
+  done
+done
+
 
 # Capture failures
 EXIT_STATUS=0 # everything passed
 function set_failed_status {
   EXIT_STATUS=1
 }
 
-
 # Setup service account credentials.
 export GOOGLE_APPLICATION_CREDENTIALS=${KOKORO_GFILE_DIR}/service-account.json
 
 case $JOB_TYPE in
-test)
-  (bundle update && bundle exec rake circleci:build) || set_failed_status
+presubmit)
+  cd $PACKAGE
+  for version in "${RUBY_VERSIONS[@]}"; do
+    rbenv global "$version"
+    echo "================================================="
+    echo "============= Using Ruby - $version ============="
+    echo "================================================="
+    (bundle update && bundle exec rake ci) || set_failed_status
+  done
   ;;
-post)
-  (bundle update && bundle exec rake circleci:post) || set_failed_status
+continuous)
+  cd $PACKAGE
+  (bundle update && bundle exec rake ci:acceptance) || set_failed_status
   ;;
 release)
```

Comment:
Do we have release integration? I believe this is looking for the `CIRCLE_TAG` environment variable.

Reply:
We don't fully yet. That's something I'll be working on once we have a functioning, satisfactory setup for PR's.